### PR TITLE
more refactoring of foreach semantic

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

**Review by ibuclaw — APPROVED** [review]
Seems reasonable.

File: src/dmd/statementsem.d [hunk 1]
```diff
@@ -1899,10 +1886,10 @@ private extern (C++) final class StatementSemanticVisitor : Visitor
         ec = new VarExp(Loc.initial, fdapply, false);
         ec = new CallExp(loc, ec, fs.aggr, fexp);
         ec.type = Type.tint32; // don't run semantic() on ec
-        return ec;
+        return cast(CallExp)ec;
```
Comment (ibuclaw) [inline src/dmd/statementsem.d:1889]: Or you could create a new variable for returning, following the unwritten "set once" rule for temporaries.

Comment (thewilsonator) [reply src/dmd/statementsem.d:1889]: Its not  temporary, see the line above.

Comment (ibuclaw) [reply src/dmd/statementsem.d:1889]: I meant locals in general.

**Comment by WalterBright** [comment]
3 hours ago I asked you to stop doing multiple refactoring commits, and 2 hours ago another one is merged?

**Pull request merged**